optimizely/optimizely.py

```diff
@@ -63,51 +70,41 @@ def __init__(self,
       self.logger.exception(str(error))
       return
 
-    error_msg = None
-    try:
-      self.config = project_config.ProjectConfig(datafile, self.logger, self.error_handler)
-    except exceptions.UnsupportedDatafileVersionException as error:
-      error_msg = error.args[0]
-      error_to_handle = error
-    except:
-      error_msg = enums.Errors.INVALID_INPUT_ERROR.format('datafile')
-      error_to_handle = exceptions.InvalidInputException(error_msg)
-    finally:
-      if error_msg:
-        self.is_valid = False
-        # We actually want to log this error to stderr, so make sure the logger
-        # has a handler capable of doing that.
-        self.logger = _logging.reset_logger(self.logger_name)
-        self.logger.exception(error_msg)
-        self.error_handler.handle_error(error_to_handle)
-        return
+    if not self.config_manager:
```

Comment:
Does it make sense to combine this with the `sdk_key` check? There doesn't seem to be additional logic outside of the inner `if` clause